python/ray/autoscaler/aws/defaults.yaml

```diff
@@ -43,38 +39,44 @@ auth:
 # configurations below.
 #    ssh_private_key: /path/to/your/key.pem
 
-# Provider-specific config for the head node, e.g. instance type. By default
-# Ray will auto-configure unspecified fields such as SubnetId and KeyName.
-# For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+# Tell the autoscaler the allowed node types and the resources they provide.
+# The key is the name of the node type, which is just for debugging purposes.
+# The node config specifies the launch config and physical instance type.
+available_node_types:
+    ray.head.default:
+        node_config:
+          InstanceType: m5.large
+          # You can provision additional disk space with a conf as follows
+          BlockDeviceMappings:
+              - DeviceName: /dev/sda1
+                Ebs:
+                    VolumeSize: 100
+        resources: {"CPU": 2}
+        min_workers: 0
+        max_workers: 0
+    ray.worker.small:
+        node_config:
+          InstanceType: m5.large
+          # Run workers on spot by default. Comment this out to use on-demand.
+          InstanceMarketOptions:
+              MarketType: spot
+        # Autoscaler will auto fill the CPU resources below.
+        resources: {"CPU": 2}
+        min_workers: 0
+        max_workers: 1
+
+# Specify the node type of the head node (as configured above).
+head_node_type: ray.head.default
+
+# The default settings for the head node. This will be merged with the per-node
+# type configs given above.
 head_node:
-    InstanceType: m5.large
-    ImageId: ami-0a2363a9cff180a64 # Deep Learning AMI (Ubuntu) Version 30
-
-    # You can provision additional disk space with a conf as follows
-    BlockDeviceMappings:
-        - DeviceName: /dev/sda1
-          Ebs:
-              VolumeSize: 100
-
-    # Additional options in the boto docs.
+    ImageId: latest_dlami
 
-# Provider-specific config for worker nodes, e.g. instance type. By default
-# Ray will auto-configure unspecified fields such as SubnetId and KeyName.
-# For more documentation on available fields, see:
-# http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.ServiceResource.create_instances
+# The default settings for worker nodes. This will be merged with the per-node
+# type configs given above.
 worker_nodes:
-    InstanceType: m5.large
-    ImageId: ami-0a2363a9cff180a64 # Deep Learning AMI (Ubuntu) Version 30
-
-    # Run workers on spot by default. Comment this out to use on-demand.
-    InstanceMarketOptions:
-        MarketType: spot
-        # Additional options can be found in the boto docs, e.g.
-        #   SpotOptions:
-        #       MaxPrice: MAX_HOURLY_PRICE
-
-    # Additional options in the boto docs.
+    ImageId: latest_dlami
```

Comment:
didn't we want to get rid of this cc @ijrsvt

Reply:
@richardliaw 
Is this discussion relevant to this PR?

For now, I am just making sure the defaults are the same as example multi node type. If you think we should change the image defaults, it can be done in a separate PR.

Reply:
@AmeerHajAli you're introducing a new ImageId to the defaults - I think either you should keep the status quo, or open the floor to discussion as to which ImageId should be default.

Reply:
@richardliaw this isn't new behavior, this is a no-op. We've always defaulted to the deep learning AMI, it's just programmatically filled in if the field isn't included in this yaml. 

https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/aws/config.py#L501

Reply:
wtf... that's confusing